.github/workflows/unit_test.yml

```diff
@@ -3,12 +3,16 @@ name: Run unit tests
 on:
   workflow_call:
     inputs:
+      ref:
+        type: string
+        default: ''
+        required: true
       runner:
         type: string
         default: '["base", "dind", "2204"]'
       runner-archs:
         type: string
-        default: '["amd64"]'
+        default: '["amd64", "arm64"]'
```

Comment:
I am not sure if testing in aarch64 is needed for unit testing.

Reply:
I am afraid it is needed. Differences in endianness, alignment requirements, or atomic operations can cause subtle bugs. Even if the code is architecture-neutral as in the case of Go, underlying libraries or system calls might behave differently. Also, cross-compilation is not equivalent to cross-testing. Just compiling for arm64 on an amd64 host doesn’t guarantee runtime correctness on arm64.

We could definitely skip for now if we want to minimize test time, but it is advisable that we introduce unit testing for arm64 as well in the near future.